subsys/bluetooth/host/id.c

```diff
@@ -353,6 +353,11 @@ int bt_id_set_private_addr(uint8_t id)
 
 	__ASSERT_NO_MSG(id < CONFIG_BT_ID_MAX);
 
+	/* bt_rand is not available before Bluetooth enable has been called */
+	if (!atomic_test_bit(bt_dev.flags, BT_DEV_ENABLE)) {
+		return -EINVAL;
```

Comment:
Perhaps `EAGAIN` is a better return value, as technically the arguments are not invalid. 

Applies multiple places.